### PR TITLE
Add SpatialRef::to_projjson()

### PR DESCRIPTION
**Pull request by ttencate**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: src/spatial_ref/srs.rs [hunk 1]
```diff
@@ -506,6 +506,22 @@ impl SpatialRef {
...
+        let rv = unsafe { gdal_sys::OSRExportToPROJJSON(self.0, &mut c_projjsonstr, options) };
+        let res = if rv != OGRErr::OGRERR_NONE {
+            Err(GdalError::OgrError {
+                err: rv,
+                method_name: "OSRExportToPROJJSON",
+            })
+        } else {
+            Ok(_string(c_projjsonstr))
+        };
+        unsafe { gdal_sys::VSIFree(c_projjsonstr.cast::<std::ffi::c_void>()) };
```
Comment (lnicola) [inline src/spatial_ref/srs.rs:521]: I don't even know any more which one of `.cast()`, `.cast<c_void>()` or `as _` is preferred :smiley:. At least this is consistent with the rest of the code.

File: src/spatial_ref/srs.rs [hunk 2]
```diff
@@ -506,6 +506,22 @@ impl SpatialRef {
         res
     }
 
+    pub fn to_projjson(&self) -> Result<String> {
```
Comment (lnicola) [inline src/spatial_ref/srs.rs:509]: I hate this spelling, TBH, but...

**Review by lnicola — APPROVED** [review]

**Comment by lnicola** [comment]
Looks like we're still testing GDAL 3.0.4 on CI, and `OSRExportToPROJJSON` was added in 3.1.

File: src/spatial_ref/srs.rs (hunk 1 above)
Comment (ttencate) [reply src/spatial_ref/srs.rs:521]: Yeah, it's mostly a copy/paste... The entire lib could do with some helper functions/macros to streamline stuff like this.

File: src/spatial_ref/srs.rs (hunk 2 above)
Comment (ttencate) [reply src/spatial_ref/srs.rs:509]: Me too, but it's consistent...

**Comment by lnicola** [comment]
Please drop that PAM file.

bors d+

**Comment by ttencate** [comment]
bors r+

Thanks for the quick turnaround!

File: src/spatial_ref/srs.rs [hunk 3]
```diff
@@ -506,6 +506,23 @@ impl SpatialRef {
         res
     }
 
+    #[cfg(any(major_ge_4, all(major_ge_3, minor_ge_1)))]
+    pub fn to_projjson(&self) -> Result<String> {
+        let mut c_projjsonstr = ptr::null_mut();
+        let options = ptr::null();
```
Comment (lnicola) [inline src/spatial_ref/srs.rs:512]: In the long run, I'd like us to support options where applicable:

https://gdal.org/doxygen/classOGRSpatialReference.html#a1a4b65b551d3a5ea91fb7fd8d9e1e4e8

**Comment by lnicola** [comment]
`tinymarble.tif.aux.xml` strikes again!

**Comment by lnicola** [comment]
You've added the PAM file again when changing the assert message.

**Comment by ttencate** [comment]
Just added it to `.gitignore` so it won't bother anyone again.

**Comment by lnicola** [comment]
Well, I was planning to fix `TempFixture` or the test to clean up better after itself :slightly_smiling_face:.

**Comment by ttencate** [comment]
> Well, I was planning to fix `TempFixture` or the test to clean up better after itself slightly_smiling_face.

That's great, but let's keep this thing from wasting everyone's time until you get round to it :)

**Pull request merged**